enclave_manager/avalon_enclave_manager/enclave_manager.py

```diff
@@ -473,33 +497,70 @@ def start_enclave_manager(config):
                      "exiting SGX Enclave manager: {err}")
         exit(1)
 
-    try:
-        sleep_interval = int(config["EnclaveManager"]["sleep_interval"])
-    except Exception as err:
-        logger.error("Failed to get sleep interval from config file. " +
-                     "Setting sleep interval to 10 seconds: %s", str(err))
-        sleep_interval = 10
+    if int(config["WorkloadExecution"]["sync_workload_execution"]) == 1:
+        # Binding with ZMQ Port
+        try:
+            socket = bind_zmq_socket(config.get("EnclaveManager")["zmq_port"])
+            logger.info("ZMQ Port hosted by Enclave")
+        except Exception:
+            logger.exception("Failed to bind socket")
 
-    try:
-        while True:
-            # Poll KV storage for new work-order requests and process
-            enclave_manager.process_work_orders(kv_helper)
-            logger.info("Enclave manager sleeping for %d secs", sleep_interval)
-            time.sleep(sleep_interval)
-    except Exception as inst:
-        logger.error("Error while processing work-order; " +
-                     "shutting down enclave manager")
-        logger.error("Exception: {} args {} details {}".format(type(inst),
-                     inst.args, inst))
-        exit(1)
+        try:
+            while True:
+                # Wait for the next request
+                logger.info("Enclave Manager waiting for next request")
+                wo_id = socket.recv()
+                logger.info("Received request at enclave manager: %s" % wo_id)
+                result = enclave_manager.process_work_order_sync(kv_helper,
+                                                                 wo_id)
+                if result is None:
+                    socket.send_string("Error while processing work order")
```

Comment:
Can you also send wo_id along with message? It would be helpful in knowing which work order execution has failed.

Reply:
Updated